dotCMS/src/main/java/com/dotmarketing/portlets/templates/business/TemplateAPI.java

```diff
@@ -193,16 +193,33 @@ Template copy(Template sourceTemplate, Host destinationHost, boolean forceOverwr
 	 */
 	void unarchive (Template template);
 
+	/**
+	 * Delete the specified template.
+	 *
+	 * Template should be archive, otherwise DotStateException
+	 * Should have write permission, otherwise SecurityException
+	 *
+	 * @param template {@link Template}
+	 * @param user     {@link User}
+	 * @param respectFrontendRoles {@link Boolean}
+	 * @return boolean
+	 * @throws DotSecurityException
+	 * @throws Exception
+	 */
+	public boolean deleteTemplate(Template template, User user, boolean respectFrontendRoles);
```

Comment:
Codacy found an issue: [Unnecessary modifier 'public' on method 'deleteTemplate': the method is declared in an interface type](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6309437)